internal/planner/min.go

```diff
@@ -0,0 +1,233 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package planner
+
+import (
+	"math"
+
+	"github.com/sourcenetwork/immutable"
+
+	"github.com/sourcenetwork/defradb/client/request"
+	"github.com/sourcenetwork/defradb/internal/core"
+	"github.com/sourcenetwork/defradb/internal/planner/mapper"
+)
+
+type minNode struct {
+	documentIterator
+	docMapper
+
+	p    *Planner
+	plan planNode
+
+	isFloat           bool
+	virtualFieldIndex int
+	aggregateMapping  []mapper.AggregateTarget
+
+	execInfo minExecInfo
+}
+
+type minExecInfo struct {
+	// Total number of times minNode was executed.
+	iterations uint64
+}
+
+func (p *Planner) Min(
+	field *mapper.Aggregate,
+	parent *mapper.Select,
+) (*minNode, error) {
+	isFloat := false
+	for _, target := range field.AggregateTargets {
+		isTargetFloat, err := p.isValueFloat(parent, &target)
+		if err != nil {
+			return nil, err
+		}
+		// If one source property is a float, the result will be a float - no need to check the rest
+		if isTargetFloat {
+			isFloat = true
+			break
+		}
+	}
+
+	return &minNode{
+		p:                 p,
+		isFloat:           isFloat,
+		aggregateMapping:  field.AggregateTargets,
+		virtualFieldIndex: field.Index,
+		docMapper:         docMapper{field.DocumentMapping},
+	}, nil
+}
+
+func (n *minNode) Kind() string           { return "minNode" }
+func (n *minNode) Init() error            { return n.plan.Init() }
+func (n *minNode) Start() error           { return n.plan.Start() }
+func (n *minNode) Spans(spans core.Spans) { n.plan.Spans(spans) }
+func (n *minNode) Close() error           { return n.plan.Close() }
+func (n *minNode) Source() planNode       { return n.plan }
+func (n *minNode) SetPlan(p planNode)     { n.plan = p }
+
+func (n *minNode) simpleExplain() (map[string]any, error) {
+	sourceExplanations := make([]map[string]any, len(n.aggregateMapping))
+
+	for i, source := range n.aggregateMapping {
+		simpleExplainMap := map[string]any{}
+
+		// Add the filter attribute if it exists.
+		if source.Filter == nil {
+			simpleExplainMap[filterLabel] = nil
+		} else {
+			// get the target aggregate document mapping. Since the filters
+			// are relative to the target aggregate collection (and doc mapper).
+			var targetMap *core.DocumentMapping
+			if source.Index < len(n.documentMapping.ChildMappings) &&
+				n.documentMapping.ChildMappings[source.Index] != nil {
+				targetMap = n.documentMapping.ChildMappings[source.Index]
+			} else {
+				targetMap = n.documentMapping
+			}
+			simpleExplainMap[filterLabel] = source.Filter.ToMap(targetMap)
```

Comment:
suggestion: Similar comment as made on `max` explain.

Reply:
should also be covered now